flowey/flowey_lib_common/src/ado_task_publish_test_results.rs

```diff
@@ -0,0 +1,77 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+//! ADO Task Wrapper: `PublishTestResults@2`
+
+use flowey::node::prelude::*;
+
+#[derive(Serialize, Deserialize)]
+pub enum AdoTestResultsFormat {
+    JUnit,
+    NUnit,
+    VSTest,
+    XUnit,
+    CTest,
+}
+
+impl std::fmt::Display for AdoTestResultsFormat {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        match self {
+            AdoTestResultsFormat::JUnit => write!(f, "JUnit"),
+            AdoTestResultsFormat::NUnit => write!(f, "NUnit"),
+            AdoTestResultsFormat::VSTest => write!(f, "VSTest"),
+            AdoTestResultsFormat::XUnit => write!(f, "XUnit"),
+            AdoTestResultsFormat::CTest => write!(f, "CTest"),
+        }
+    }
+}
+
+flowey_request! {
+    pub struct Request {
+        pub step_name: String,
+        pub format: AdoTestResultsFormat,
+        pub files: ReadVar<String>,
```

Comment:
If you think about these backend-specific wrapper nodes as a form of "FFI", I would suggest having these nodes expose a strongly-typed Rust API, and then doing any to/from String convertions within the node itself. so, in this case, maybe `files` is a `Vec<PathBuf>`

Reply:
The files input can accept a bunch of interesting stuff ([docs](https://learn.microsoft.com/en-us/azure/devops/pipelines/tasks/reference/publish-test-results-v2?view=azure-pipelines&tabs=trx%2Ctrxattachments%2Cyaml#:~:text=testResultsFiles%20%2D%20Test%20results%20files)) but I suppose I can simplify the api to only give it actual path names